### ISSUE TITLE
Kafka-Node Crashes Index Out of Range (client.js)

### ISSUE THREAD

**Issue by zach-snell** [SOHU-Co/kafka-node#669]
## Bug Report

Kafka Node crashes after duration while idle (no in/out data flow).

Expected: 
The application can run idle for infinite duration.

Actual: 
Application is crashing while idle after a duration ranging from 5 mins to 2 hours.

Reproduction:
Spin up Kafka (I used docker spotify/kafka)
Generate a topic ( I created 2 with two attached producers, one console and one kafka-node. )
Connect with a consumer on a generated topic from kafka-node.
Idle until crash

### Environment

* Node version: v6.9.4
* Kafka-node version: v1.6.0
* Kafka version: 2.11-0.10.1.0

### For specific cases also provide 
(using docker spotify/kafka)
* Number of Brokers: default
* Number partitions for topic: default

### Include Sample Code to reproduce behavior

This issue occurs after a random idle duration between 5 mins and 120 mins

Focusing on this code: 
```js
Client.prototype.handleReceivedData = function (socket) {
  var buffer = socket.buffer;
  if (!buffer.length) {
    return;
  }
  // Watch for a buffer to come through as readUInt8
  var size = buffer.readUInt32BE(0) + 4; // EXCEPTION, index out of bounds (I think due to there being no UInt32 in the _buffs of the buffer
  ...

}
```

### Include output with [Debug](https://github.com/SOHU-Co/kafka-node#how-do-i-debug-an-issue) turned on

RangeError: Index out of range
    at checkOffset (buffer.js:817:11)
    at Buffer.readUInt32BE (buffer.js:891:5)
    at BufferList.(anonymous function) [as readUInt32BE] (/Users/zachsnell/develop/realm-event-handler-poc/node_modules/bl/bl.js:273:58)
    at Client.handleReceivedData (/Users/zachsnell/develop/realm-event-handler-poc/node_modules/kafka-node/lib/client.js:735:21)
    at Socket.<anonymous> (/Users/zachsnell/develop/realm-event-handler-poc/node_modules/kafka-node/lib/client.js:700:10)
    at emitOne (events.js:96:13)
    at Socket.emit (events.js:188:7)
    at readableAddChunk (_stream_readable.js:176:18)
    at Socket.Readable.push (_stream_readable.js:134:10)
    at TCP.onread (net.js:548:20)

**Comment by hyperlink** [comment]
Looks like you're using 1.6.1 since it's using buffer list.  Does this happen with 1.6.0?

**Comment by zach-snell** [comment]
@hyperlink  You are right, I very lazily added this to my package.json and allowed the newer version. I will try with 1.6.0 and let you know the result.

Thanks!

**Comment by hyperlink** [comment]
btw I'm working on 1.6.2 release with those breaking changes reverted.

**Comment by hyperlink** [comment]
Published as 1.6.2

**Comment by crzidea** [comment]
OK, I get it! The condition should not be `!buffer.length`, it should be `buffer.length < 4`. @hyperlink There is no need to revert the `bl` change completely.
